zerver/models/realms.py

```diff
@@ -826,6 +825,27 @@ def max_invites(self) -> int:
     def max_invites(self, value: Optional[int]) -> None:
         self._max_invites = value
 
+    @property
+    def upload_quota_gb(self) -> Optional[int]:
+        # See upload_quota_bytes; don't interpret upload_quota_gb directly.
+
+        if self.custom_upload_quota_gb is not None:
+            return self.custom_upload_quota_gb
+
+        plan_type = self.plan_type
+        if plan_type == Realm.PLAN_TYPE_PLUS:
+            return Realm.UPLOAD_QUOTA_STANDARD
+        elif plan_type == Realm.PLAN_TYPE_STANDARD:
+            return Realm.UPLOAD_QUOTA_STANDARD
+        elif plan_type == Realm.PLAN_TYPE_SELF_HOSTED:
+            return None
+        elif plan_type == Realm.PLAN_TYPE_STANDARD_FREE:
+            return Realm.UPLOAD_QUOTA_STANDARD
+        elif plan_type == Realm.PLAN_TYPE_LIMITED:
+            return Realm.UPLOAD_QUOTA_LIMITED
+        else:
+            raise AssertionError("Invalid plan type")
+
     def upload_quota_bytes(self) -> Optional[int]:
```

Comment:
So while reading this I noticed that `upload_quota_bytes` is written to the API as `upload_quota_mb` -- i..e there's a factor of 1M bug in it. Probably we should fix `zerver/lib/events.py` to compute the correct value ... and also check the web app uploads panel for whether it needs a corresponding fix.

(I don't think mobile clients read this field).

Reply:
@timabbott Push a prep commit that fixes it and tweaks the webapp to correctly convert the new values.